File: algorithms/hackerrank/algorithms/warmup/PlusMinus.java
import java.util.Scanner;

public class PlusMinus {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);

        int n = in.nextInt();

        int negative = 0;
        int positive = 0;
        int zero = 0;

        for (int i = 0; i < n; i++) {
            int tmp = in.nextInt();

            if (tmp > 0) {
                positive++;
            } else if (tmp < 0) {
                negative++;
            } else {
                zero++;
            }
        }

        System.out.format("%.6f\n", (double) positive / n);
        System.out.format("%.6f\n", (double) negative / n);
        System.out.format("%.6f\n", (double) zero / n);
    }
}

File: algorithms/hackerrank/algorithms/warmup/plus-minus.go
package main

import "fmt"

func main() {
	var k int
	var t int

	var positive float32 = 0
	var negative float32 = 0
	var zeros float32 = 0

	fmt.Scan(&k)

	for i := 0; i < k; i++ {
		fmt.Scan(&t)

		if t == 0 {
			zeros += 1
		} else if t > 0 {
			positive += 1
		} else if t < 0 {
			negative += 1
		}
	}

	fmt.Printf("%.06f\n%.06f\n%.06f", positive / float32(k), negative / float32(k), zeros / float32(k))
}
